pkg/storage/mongo/saver.go

```diff
@@ -23,10 +24,14 @@ func (s *ModuleStore) Save(ctx context.Context, module, version string, mod []by
 	}
 	defer f.Close()
 
-	_, err = io.Copy(f, zip) // check number of bytes written?
+	numBytesWritten, err := io.Copy(f, zip)
 	if err != nil {
 		return errors.E(op, err, errors.M(module), errors.V(version))
 	}
+	if numBytesWritten <= 0 {
+		e := fmt.Errorf("copied %d bytes to Mongo GridFS", numBytesWritten)
```

Comment:
I think since this only fires when there were `<=0` bytes written we could just write than in.

Reply:
looks good, i wonder if we can get content length from vcs response and use it here for validation, not needed to be done in this PR (separate ok) just a thought

Reply:
oh we're not using API calls so we cannot but we can do Stat to find out filesize and pass it down,
in case we start using API calls we can do the same with content length